### ISSUE TITLE
PA Review: User Authentication

### ISSUE THREAD

**Issue by ThinkSalat** [ThinkSalat/Trendr#2]
- [x] Backend: DB, model, controller for Users
- [x] Redux Loop: ajax, actions, reducer for Users, Session
- [x] Presentational Components and Containers for Header, Session Pages
- [x] Styling 
- [x] Smooth, bug-free navigation
- [x] Demo User logs in

**Comment by andywynkoop** [comment]
- [ ] Remove redux logger and all console logs in production
- [ ] When I tried to sign up again using the same credentials, I got a blank screen and the console says `this.props.errors is not a function`. Only seems to happen if I created that account in the same session.
- [ ] Going to a random route `/gdsbsdfsgv` should redirect or render a 404

**Comment by andywynkoop** [comment]
A few small changes:
- [x] Remove redux logger and all console logs in production
- [x] When I tried to sign up again using the same credentials, I got a blank screen and the console says `this.props.errors is not a function`. Only seems to happen if I created that account in the same session.
- [x] Going to a random route `/gdsbsdfsgv` should redirect or render a 404
- [x] All clickable things should be styled with `cursor: pointer`; login, signup buttons, logout button, etc.
